probe-rs/src/bin/probe-rs/cmd/complete.rs

```diff
@@ -0,0 +1,219 @@
+use std::io::{Cursor, Write};
+
+use anyhow::{anyhow, Context};
+use clap::CommandFactory;
+use clap_complete::{generate, Shell};
+use probe_rs::probe::list::Lister;
+
+use crate::{util::common_options::OperationError, Cli};
+
+const BIN_NAME: &str = "probe-rs";
+
+/// Install and complete autocomplete scripts
+#[derive(clap::Parser)]
+#[clap(verbatim_doc_comment)]
+pub struct Cmd {
+    #[clap(long)]
+    shell: Option<Shell>,
+    #[clap(subcommand)]
+    kind: CompleteKind,
+}
+
+impl Cmd {
+    pub fn run(&self, lister: &Lister) -> Result<(), anyhow::Error> {
+        let shell = Shell::from_env()
+            .or(self.shell)
+            .ok_or_else(|| anyhow!("The current shell could not be determined. Please specify a shell with the --shell argument."))?;
+        if !matches!(shell, Shell::Zsh | Shell::Bash) {
```

Comment:
Can we also have PowerShell in the list, maybe without the dynamic completions for now?